features/sps/process.feature

```diff
@@ -1,11 +1,52 @@
+
 Feature: Science Processing Service
+  The ADES is fronted by a WPS-T service which allows for the deployment and
+  undeployment of application packages. It also allows for the exectuion,
+  status, and retrieval of job submissions. This is captured in the zenhub epic
+  at https://app.zenhub.com/workspaces/unity-workspace-61d4b6a48ed26b001d7d184a/issues/unity-sds/unity-sps-prototype/57
 
-  Scenario: Request an Aglorithm Deployment
+  Scenario: Request an Algorithm Deployment
 
   Scenario: Undeploy an Algorithm deployment
 
+  Scenario: List deployed WPS Processes
+    Given the SoundsSIPS L1A algorithm has been deployed to the ADES
+    And the SoundsSIPS L1B algorithm has been deployed to the ADES
+    When GetCapabilities is called on the WPS-T endpoint
+    Then the WPS-T response is an http-200 Status
+    And the response includes process summary elements
+    And the process summary included the L1A and L1B processors
+
   Scenario: Get the inputs for a given Algorithm deployment
+    Given the SoundsSIPS L1A algorithm has been deployed to the ADES
+    When DescribeProcess is called on the WPS-T endpoint for the L1A Algorithm
+    Then the WPS-T endpoint responds with a ProcessOfferings response
+    And the response page returns a HTTP 200
+    And the response page includes the one or more input element
+    # more to add here?
+
+  Scenario: Request L1A Processing from an Algorithm deployment
+    Given the SoundsSIPS L1A algorithm has been deployed to the ADES
+    And SounderSIPS L0 data exists for a 2 hour block of Data in the Unity system
```

Comment:
Presumably this embeds some knowledge of the L1A algorithm... is there no way to query SPS/WPS-T directly to say "are all the requirements met for this algorithm to run"?

Reply:
this is a good thread to pull on, and i think its important to think of generic testing in the future, but we're not testing all of the input/output variants of the WPS here, we're testing the major interfaces. Unit/integration testing on the WPS-T deployment should cover some generic testing, and the projects wil need to bring their own validation at some point. there is a fuzzy area where we could write some validation, and that's probably valuable from a user sanity perspective, just not right now.